### ISSUE TITLE
3.6 wheels no longer fixable by auditwheel

### ISSUE THREAD

**Issue by AvdN** [pypa/manylinux#214]
First of all thank you for manylinux, I use it to make the Linux wheels for `ruamel.yaml`. 

Until Python 3.7 came out I was using ``quay.io/pypa/manylinux1_x86_64:01a75168a06f``  (which was the latest version early 2017. That generated my 2.7, 3.4, 3.5 and 3.6 wheels.
After upgrading last month to 48b332d3a997 I could generate the 3.7 wheels, but failed to notice that 3.6 no long is generated.  3 days ago I update to fbf76948b80e, which did not solve the issue.

I added ``auditwheel show`` to the script that runs within the Docker container, this generates the same (bar python version numbers) output for 2.7, 3.4, 3.5 as for 3.7 (which is included below) but not the same for 3.6 (I stripped the container name from every line):

```
processing /tmp/ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl

ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl is consistent with the
following platform tag: "linux_x86_64".

The wheel references external versioned symbols in these system-
provided shared libraries: libc.so.6 with versions {'GLIBC_2.14',
'GLIBC_2.2.5', 'GLIBC_2.4', 'GLIBC_2.3.4'}

This constrains the platform tag to "linux_x86_64". In order to
achieve a more compatible tag, you would need to recompile a new wheel
from source on a system with earlier versions of these libraries, such
as CentOS 5.
Repairing ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl
usage: auditwheel [-h] [-V] [-v] command ...
auditwheel: error: cannot repair "/tmp/ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl" to "manylinux1_x86_64" ABI because of the presence of too-recent versioned symbols. You'll need to compile the wheel on an older toolchain.
processing /tmp/ruamel.yaml-0.15.45-cp37-cp37m-linux_x86_64.whl

ruamel.yaml-0.15.45-cp37-cp37m-linux_x86_64.whl is consistent with the
following platform tag: "manylinux1_x86_64".

The wheel references external versioned symbols in these system-
provided shared libraries: libc.so.6 with versions {'GLIBC_2.2.5'}

The following external shared libraries are required by the wheel:
{
    "libc.so.6": "/lib64/libc-2.5.so",
    "libpthread.so.0": "/lib64/libpthread-2.5.so"
}
Repairing ruamel.yaml-0.15.45-cp37-cp37m-linux_x86_64.whl
Previous filename tags: linux_x86_64
New filename tags: manylinux1_x86_64
Previous WHEEL info tags: cp37-cp37m-linux_x86_64
New WHEEL info tags: cp37-cp37m-manylinux1_x86_64

Fixed-up wheel written to /src/dist/ruamel.yaml-0.15.45-cp37-cp37m-manylinux1_x86_64.whl
```

The Python 3.6 seems to reference a lot more GLIBC versions then the others,  who all just use GLBOC_2.2.5 and can be auditwheeled fine.

I changed the call to the script to only build the 3.6 wheel, to see if there was some garbage from the other builds that influences this, but that did not help.

Fortunately I did have older images in my image cache ``docker images  -a | fgrep quay``:

```
quay.io/pypa/manylinux1_x86_64            latest               fbf76948b80e        3 days ago          881MB
quay.io/pypa/manylinux1_x86_64            <none>               48b332d3a997        4 weeks ago         1.16GB
quay.io/pypa/manylinux1_x86_64            <none>               01a75168a06f        18 months ago       1.21GB
```

And changing my Dockerfile to start with ``FROM quay.io/pypa/manylinux1_x86_64:01a75168a06f`` allowed me to make 3.6 wheels. 

The output still mentions multiple GLIBC versions byt auditwheel doesn't error out:

```
 processing /tmp/ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl
 
 ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl is consistent with the
 following platform tag: "manylinux1_x86_64".
 
 The wheel references external versioned symbols in these system-
 provided shared libraries: libc.so.6 with versions {'GLIBC_2.4',
 'GLIBC_2.3.4', 'GLIBC_2.14', 'GLIBC_2.2.5'}
 
 The following external shared libraries are required by the wheel:
 {
     "libc.so.6": "/lib64/libc-2.5.so",
     "libpthread.so.0": "/lib64/libpthread-2.5.so"
 }
 Repairing ruamel.yaml-0.15.45-cp36-cp36m-linux_x86_64.whl
 Previous filename tags: linux_x86_64
 New filename tags: manylinux1_x86_64
 Previous WHEEL info tags: cp36-cp36m-linux_x86_64
 New WHEEL info tags: cp36-cp36m-manylinux1_x86_64
 
 Fixed-up wheel written to /src/dist/ruamel.yaml-0.15.45-cp36-cp36m-manyli
```

I am not sure if that 01a75168a06f image  can still be downloaded from Docker (in case my cache gets cleaned), nor have I investigated that I can explicitly back it up. 

I can work with two Dockerfiles if necessary, but switching back-and-forth between Docker images is not as comfortable as starting docker once and generating all wheels in one go.

**Comment by AvdN** [comment]
The version for 3.6 that I now make with the old Docker image seems to have some problems in not removing RPATH (https://bitbucket.org/ruamel/yaml/issues/212/invalid-rpath-in-latest-release). I checked the output of the run for 3.6 once more and it seems ok.

And now of course I really wonder how that can happen when using the same Docker image (and auditwheel) as before.

**Comment by AvdN** [comment]
I found the cause for this. After adding code on the host to remove the RPATH in the .so to work around the issue mentioned in the previous comment, I noticed that that .so had a date of July 8th.

I map my host source directory onto '/src' on the and there still was a build directory with only one file: the .so for 3.6. I am not sure why building did not overwrite this, and why auditwheel did not notice that its actions did not work. 

After removing the ``build`` directory, I was able to generate 2.7/3.4/3.5/3.6/3.7 in one run with the latest manylinux1 Docker images.

My apologies for the confusion
